Factories/CMakeLists.txt

```diff
@@ -0,0 +1,101 @@
+cmake_minimum_required (VERSION 2.6)
+project (histFactory)
+
+# Configure paths
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake/modules")
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/../cmake/modules")
+
+# Detect if we are inside a CMSSW env
+include(CMSSW)
+
+# Ensure C++11 is available
+include(CheckCXXCompilerFlag)
+CHECK_CXX_COMPILER_FLAG("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
+if(COMPILER_SUPPORTS_CXX0X)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -std=c++0x -g")
+else()
+    message(STATUS "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support. Please use a different C++ compiler.")
+endif()
+
+# Find ROOT
+find_package(ROOT REQUIRED)
```

Comment:
`GenVector` is already linked to the plotter / skimmer. See `templates/CMakeLists.txt.tpl`